connectors/sources/sharepoint_online.py

```diff
@@ -867,6 +880,13 @@ def is_domain_group(user_fields):
     )
 
 
+def is_dynamic_group(login_name):
+    if not login_name:
+        return False
+
+    return login_name.startswith("c:0o.c|federateddirectoryclaimprovider")
```

Comment:
I don't think this is indicative of dynamic groups, but just of "groups". See: https://github.com/elastic/connectors-python/blob/main/connectors/sources/sharepoint_online.py#L862-L867

Reply:
It seems it's the combination of `odata.type` being `SP.User` and the `LoginName` starting with `c:0o.c|federateddirectoryclaimprovider`. We can probably keep it local to the caller method. In the method's context it makes more sense.